specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/StackHCI/edgeMachine.tsp

```diff
@@ -0,0 +1,813 @@
+}
+
+/**
+ * This represents extensions installed on virtualSwitch.
+ */
+@added(Versions.v2025_12_01_preview)
+model SwitchExtension {
+  /**
+   * Unique identifier for virtualSwitch.
+   */
+  @visibility(Lifecycle.Read)
+  switchId?: string;
+
+  /**
+   * This will show extension name for virtualSwitch.
+   */
+  @visibility(Lifecycle.Read)
+  extensionName?: string;
+
+  /**
+   * This represents whether extension is enabled on virtualSwitch.
+   */
+  @visibility(Lifecycle.Read)
+  extensionEnabled?: boolean;
+}
+
+/**
+ * OS configurations for HCI device.
+ */
+@added(Versions.v2025_12_01_preview)
+model OsProfile {
+  /**
+   * The boot type of the device. e.g. UEFI, Legacy etc
+   */
+  @visibility(Lifecycle.Read)
+  bootType?: string;
+
+  /**
+   * Version of assembly present on device
+   */
+  @visibility(Lifecycle.Read)
+  assemblyVersion?: string;
+
+  /**
+   * OS type (“windows", “linux”)
+   */
+  @visibility(Lifecycle.Read)
+  osType?: string;
+
+  /**
+   * OS SKU (e.g., “ Microsoft Azure Linux ROE“, “Azure Stack HCI", "Microsoft Azure Linux 3.0")
+   */
+  @visibility(Lifecycle.Read)
+  osSku?: string;
+
+  /**
+   * OS Version
+   */
+  @visibility(Lifecycle.Read)
+  osVersion?: string;
+
+  /**
+   * OS Build Number
+   */
+  @visibility(Lifecycle.Read)
+  buildNumber?: string;
+
+  /**
+   * OS Base Image Version
+   */
+  @visibility(Lifecycle.Read)
+  baseImageVersion?: string;
+
+  /**
+   * OS Image Version
+   */
+  @visibility(Lifecycle.Read)
+  imageVersion?: string;
+}
+
+/**
+ * Describes the RDMA capability of the network adapter.
+ */
+@added(Versions.v2025_12_01_preview)
+union RdmaCapability {
+  string,
+
+  /**
+   * Network Adapter on the device is RDMA Capable
+   */
+  Enabled: "Enabled",
+
+  /**
+   * Network Adapter on the device isn't RDMA Capable
+   */
+  Disabled: "Disabled",
+}
+
+/**
+ * Hardware profile for the machine
+ */
+@added(Versions.v2025_12_01_preview)
+model HardwareProfile {
+  /**
+   * Number of cpu cores in the machine
+   */
+  @visibility(Lifecycle.Read)
+  cpuCores?: int64;
+
+  /**
+   * Number of cpu sockets in the machine
+   */
+  @visibility(Lifecycle.Read)
+  cpuSockets?: int64;
+
+  /**
+   * Memory capacity of the machine
+   */
+  @visibility(Lifecycle.Read)
+  memoryCapacityInGb?: int64;
+
+  /**
+   * Model info of the machine
+   */
+  @visibility(Lifecycle.Read)
+  `model`?: string;
+
+  /**
+   * manufacturer info of the machine
+   */
+  @visibility(Lifecycle.Read)
+  manufacturer?: string;
+
+  /**
+   * Serial number of the machine
+   */
+  @visibility(Lifecycle.Read)
+  serialNumber?: string;
+
+  /**
+   * Process type of the machine
+   */
+  @visibility(Lifecycle.Read)
+  processorType?: string;
+}
+
+@added(Versions.v2025_12_01_preview)
+@doc("StorageProfile of edge machine.")
+model StorageProfile {
+  @doc("Number of storage disks in the device with $CanPool as true.")
+  @visibility(Lifecycle.Read)
+  poolableDisksCount?: int64;
```

Comment:
>poolableDisksCount [](http://example.com/codeflow?start=2&length=18)

Assuming the business logic will ensure the values is not negative

Reply:
Business logic handles the validation